core/src/main/java/fr/sncf/osrd/conflicts/Conflicts.kt

```diff
@@ -204,6 +241,119 @@ class IncrementalConflictDetectorImpl(trainRequirements: List<TrainRequirements>
         }
         return res
     }
+
+    override fun minDelayWithoutConflicts(
+        spacingRequirements: List<SpacingRequirement>,
+        routingRequirements: List<RoutingRequirement>,
+        globalMinDelay: Double
+    ): Double {
+        var minDelay = Double.POSITIVE_INFINITY
+        var hasConflict = false
+        for (spacingRequirement in spacingRequirements) {
+            if (spacingZoneRequirements[spacingRequirement.zone!!] != null) {
+                val conflictingRequirements =
+                    spacingZoneRequirements[spacingRequirement.zone!!]!!.filter {
+                        !(spacingRequirement.beginTime >= it.endTime ||
+                            spacingRequirement.endTime <= it.beginTime)
+                    }
+                if (conflictingRequirements.isNotEmpty()) {
+                    hasConflict = true
+                    val latestEndTime = conflictingRequirements.maxOf { it.endTime }
+                    minDelay = min(minDelay, latestEndTime - spacingRequirement.beginTime)
+                }
+            }
+        }
+        for (routingRequirement in routingRequirements) {
+            for (zoneReq in routingRequirement.zones) {
+                if (routingZoneRequirements[zoneReq.zone!!] != null) {
+                    val conflictingRequirements =
+                        routingZoneRequirements[zoneReq.zone!!]!!.filter {
+                            !(routingRequirement.beginTime >= it.endTime ||
+                                zoneReq.endTime <= it.beginTime)
+                        }
```

Comment:
routing requirements aren't always incompatible even when they overlap (see the lambda parameter given line 171)